app/models/abuse_report.rb

```diff
@@ -49,4 +65,23 @@ def send_report
     )
     reporter.send_report!
   end
+
+  # if the URL clearly belongs to a work (i.e. contains "works/123")
+  # make sure it isn't reported more than ABUSE_REPORTS_PER_WORK_MAX times per month
+  def work_is_not_over_reported
+    if url.match(/works\/\d+/)
+      # use "works/123/" instead of just the id to avoid confusion with chapter ids
+      work_params_only = url.match(/works\/\d+\//).to_s
+      existing_reports_total = AbuseReport.where("created_at > ? AND
+                                                 url LIKE ?",
+                                                 1.month.ago,
+                                                 "%#{work_params_only}%").
```

Comment:
So this will only match after the code has been put in as the current abuse url's do not end with a /

You could put a migration in to modify that if you wanted ?

I would also be tempted to anchor the match with a ^